sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/scan/AbstractScan.java

```diff
@@ -289,8 +298,40 @@ protected void setupAdditonalProxyConfiguration() throws ClientApiException {
         }
     }
 
+    protected void deactivateRules() throws ClientApiException {
+        OwaspZapFullRuleset fullRuleset = scanConfig.getFullRuleset();
+        DeactivatedRuleReferences deactivatedRuleReferences = scanConfig.getDeactivatedRuleReferences();
+        if (fullRuleset == null && deactivatedRuleReferences == null) {
+            return;
+        }
+        List<RuleReference> rulesReferences = deactivatedRuleReferences.getDeactivatedRuleReferences();
+        if (rulesReferences == null) {
+            return;
+        }
+
+        for (RuleReference ruleRef : rulesReferences) {
+            Rule ruleToDeactivate = fullRuleset.findRuleByRef(ruleRef.getRef());
```

Comment:
Would a Map or Set not be a better option in this case? Both have the advantage of not needing  to do a search each time.